src/ClientData/include/ClientData/ScopeInfo.h

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CLIENT_DATA_SCOPE_INFO_H_
+#define ORBIT_CLIENT_DATA_SCOPE_INFO_H_
+
+#include <string>
+
+namespace orbit_client_data {
+
+enum ScopeType : uint8_t { kOther = 0, kHookedFunction = 1, kApiScope = 2, kApiScopeAsync = 3 };
```

Comment:
Why `uint8_t`. Do you really need that? Are you aware that it automatically makes a values in `uint8_t` valid values for a `ScopeType` variable? I would assume only the defined ones should be valid, right? In that case don't define an underlying type.

Reply:
Also, why no `enum class`? I think these names should not live inside the namespace directly. You could either make it an enum class or move the enum into the struct.

Reply:
Good point, I'll make it enum class/